<launch>
    <arg name="robot_model" default="ur5" />
    <arg name="robotName" value="$(arg robot_model)_robot" />
    <arg name="rosVersion" default="noetic" />

    <param name="robotName" value="$(arg robotName)" />
    <param name="rosVersion" value="$(arg rosVersion)" />

    <!-- use common launch file and pass all arguments to it -->
    <group if="$(eval arg('robot_model') == 'ur5')">
        <include file="$(find ur_description)/launch/load_ur5.launch" pass_all_args="true" />
    </group>

    <group if="$(eval arg('robot_model') == 'ur10e')">
        <include file="$(find ur_description)/launch/load_ur10e.launch" pass_all_args="true" />
    </group>

    <node name="test_robotic_arm_ur" pkg="wp5_robotic_arms_drivers" type="test_robotic_arm_ur"
        output="screen" >
        <param name="robotName" value="$(arg robotName)" />
        <param name="rosVersion" value="$(arg rosVersion)" />
    </node>
</launch>